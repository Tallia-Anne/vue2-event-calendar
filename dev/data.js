export default function getCalendarData(d) {
  const date = new Date(d);
  const year = date.getFullYear();
  const month = date.getMonth() + 1;
  const day = date.getDate();


  const ArrayData = [
    {
      date: `${year}-${month}-${day}`,
      title: 'buy something'
    },
    {
      date: `${year}-${month}-${day}`,
      title: 'shopping'
    },
    {
      date: `${year}-${month + 1}-2`,
      title: 'remember homework'
    },
    {
      date: `${year}-${month + 1}-15`,
      title: 'music festival'
    },
    {
      date: `${year}-${month + 2}-6`,
      title: 'a course of lectures'
    }
  ];
  const ObjectData = {};

  ArrayData.forEach((item) => {
    ObjectData[item.date] = { ...item };
  });

  return {
    Array: ArrayData,
    Object: ObjectData
  };
}

// export default {
//   Array: ArrayData,
//   Object: ObjectData
// };
