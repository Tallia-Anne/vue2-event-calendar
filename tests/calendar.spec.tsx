import { shallowMount, mount, BaseWrapper } from '@vue/test-utils';
import Calendar from '../src/index.vue';
import data from '../dev/data';

interface IWrapperOptions {
  buttonPrefix: string;
  props: any;
  mountOptions?: any;
  handleMounted?: (wrapper: BaseWrapper) => void;
}

async function createDefaultWrapper({buttonPrefix, props, mountOptions, handleMounted}: IWrapperOptions) {
  const wrapper = shallowMount(Calendar, {
    ...mountOptions,
    propsData: {
      startDate: '2018-01-01',
      ...props
    }
  });

  if (!handleMounted) {
    const prev = wrapper.find(`.${buttonPrefix}-prev`);
    const next = wrapper.find(`.${buttonPrefix}-next`);
    const title = wrapper.find(`.${buttonPrefix}-header-date`);

    next.trigger('click');
    await wrapper.vm.$nextTick();
    expect(title.text()).toBe('2018-02');

    prev.trigger('click');
    await wrapper.vm.$nextTick();
    expect(title.text()).toBe('2018-01');
  } else {
    handleMounted(wrapper);
  }
}

// TODO:
// 1. test scope slots
describe('Calendar component', () => {
  test('component should match snapshot', () => {
    const wrapper = shallowMount(Calendar, {
      propsData: {
        startDate: new Date('2018-01-01'),
        dateData: data
      }
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  test('Change mode props', () => {
    const wrapper = shallowMount(Calendar, {
      propsData: {
        startDate: new Date('2018-01-01')
      }
    });

    const weekWrapper = shallowMount(Calendar, {
      propsData: {
        mode: 'week',
        startDate: new Date('2018-01-01')
      }
    });

    const monthRow = wrapper.findAll('.vue-calendar-body-row');
    const weekRow = weekWrapper.findAll('.vue-calendar-body-row');

    expect(monthRow.length).toBe(6);
    expect(weekRow.length).toBe(1);
  });

  test('has currect change action', () => {
    const prefixCls = 'kit-calendar';
    const props = { prefixCls };

    createDefaultWrapper({buttonPrefix: prefixCls, props});
  });

  test('renderHeader prop render currectly', () => {
    const props = {
      renderHeader({ prev, next, selectedDate }: any) {
        return (
          <div class='custom-header'>
            <div class='custom-header-prev' onClick={prev}>prev</div>
            <span class='custom-header-header-date'>{selectedDate}</span>
            <div class='custom-header-next' onClick={next}>next</div>
          </div>
        );
      }
    };

    createDefaultWrapper({buttonPrefix: 'custom-header', props});
  });

  test('scoped-slot should render currectly', () => {
    const component = {
      components: { Calendar },
      template: `
        <Calendar :startDate="startDate" :dateData="data">
          <div slot-scope="dateItem">
            <span>{{dateItem.date.date}} {{dateItem}}</span>
            <div
              v-for="(item, index) in dateItem.data"
              :key="index">
              {{ item.title }}
            </div>
          </div>
        </Calendar>
      `,
      data() {
        return {
          startDate: '2018-1-1',
          data: [
            {
              date: '2018-1-1',
              title: 'test scope-slot1'
            },
            {
              date: '2018-1-5',
              title: 'test scope-slot2'
            }
          ]
        };
      }
    };

    // console.log(shallow(component).element)
    expect(mount(component).element).toMatchSnapshot();
  });
});
