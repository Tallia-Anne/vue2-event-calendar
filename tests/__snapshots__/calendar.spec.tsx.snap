// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar component scoped-slot should match snapshot 1`] = `
<div
  class="calendar is-month"
>
  <div
    class="calendar-header"
  >
    <div
      class="calendar-header-left"
    />
    <div
      class="calendar-header-center"
    >
      <a
        class="calendar-control calendar-prev"
      >
        &lt;
      </a>
      <span
        class="calendar-header-date"
      >
        2018-01
      </span>
      <a
        class="calendar-control calendar-next"
      >
        &gt;
      </a>
    </div>
    <div
      class="calendar-header-right"
    />
  </div>
   
  <div
    class="calendar-week"
  >
    <div
      class="calendar-week__item"
    >
      
      Sun
    
    </div>
    <div
      class="calendar-week__item"
    >
      
      Mon
    
    </div>
    <div
      class="calendar-week__item"
    >
      
      Tue
    
    </div>
    <div
      class="calendar-week__item"
    >
      
      Wed
    
    </div>
    <div
      class="calendar-week__item"
    >
      
      Thu
    
    </div>
    <div
      class="calendar-week__item"
    >
      
      Fri
    
    </div>
    <div
      class="calendar-week__item"
    >
      
      Sat
    
    </div>
  </div>
   
  <div
    class="calendar-body"
  >
    <div
      class="calendar-body-row"
    >
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2017,
  "month": 12,
  "date": 31,
  "day": 0,
  "full": "2017-12-31"
} {
  "date": {
    "isPrevMonth": true,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": false,
    "data": [],
    "date": {
      "year": 2017,
      "month": 12,
      "date": 31,
      "day": 0,
      "full": "2017-12-31"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 1,
  "date": 1,
  "day": 1,
  "full": "2018-01-01"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": true,
    "isCurMonth": true,
    "data": [
      {
        "date": "2018-1-1",
        "title": "test scope-slot1"
      }
    ],
    "date": {
      "year": 2018,
      "month": 1,
      "date": 1,
      "day": 1,
      "full": "2018-01-01"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 1,
  "date": 2,
  "day": 2,
  "full": "2018-01-02"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": true,
    "data": [],
    "date": {
      "year": 2018,
      "month": 1,
      "date": 2,
      "day": 2,
      "full": "2018-01-02"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 1,
  "date": 3,
  "day": 3,
  "full": "2018-01-03"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": true,
    "data": [],
    "date": {
      "year": 2018,
      "month": 1,
      "date": 3,
      "day": 3,
      "full": "2018-01-03"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 1,
  "date": 4,
  "day": 4,
  "full": "2018-01-04"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": true,
    "data": [],
    "date": {
      "year": 2018,
      "month": 1,
      "date": 4,
      "day": 4,
      "full": "2018-01-04"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 1,
  "date": 5,
  "day": 5,
  "full": "2018-01-05"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": true,
    "data": [
      {
        "date": "2018-1-5",
        "title": "test scope-slot2"
      }
    ],
    "date": {
      "year": 2018,
      "month": 1,
      "date": 5,
      "day": 5,
      "full": "2018-01-05"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 1,
  "date": 6,
  "day": 6,
  "full": "2018-01-06"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": true,
    "data": [],
    "date": {
      "year": 2018,
      "month": 1,
      "date": 6,
      "day": 6,
      "full": "2018-01-06"
    }
  }
}
          </span>
           
        </div>
      </div>
    </div>
    <div
      class="calendar-body-row"
    >
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 1,
  "date": 7,
  "day": 0,
  "full": "2018-01-07"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": true,
    "data": [],
    "date": {
      "year": 2018,
      "month": 1,
      "date": 7,
      "day": 0,
      "full": "2018-01-07"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 1,
  "date": 8,
  "day": 1,
  "full": "2018-01-08"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": true,
    "data": [],
    "date": {
      "year": 2018,
      "month": 1,
      "date": 8,
      "day": 1,
      "full": "2018-01-08"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 1,
  "date": 9,
  "day": 2,
  "full": "2018-01-09"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": true,
    "data": [],
    "date": {
      "year": 2018,
      "month": 1,
      "date": 9,
      "day": 2,
      "full": "2018-01-09"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 1,
  "date": 10,
  "day": 3,
  "full": "2018-01-10"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": true,
    "data": [],
    "date": {
      "year": 2018,
      "month": 1,
      "date": 10,
      "day": 3,
      "full": "2018-01-10"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 1,
  "date": 11,
  "day": 4,
  "full": "2018-01-11"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": true,
    "data": [],
    "date": {
      "year": 2018,
      "month": 1,
      "date": 11,
      "day": 4,
      "full": "2018-01-11"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 1,
  "date": 12,
  "day": 5,
  "full": "2018-01-12"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": true,
    "data": [],
    "date": {
      "year": 2018,
      "month": 1,
      "date": 12,
      "day": 5,
      "full": "2018-01-12"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 1,
  "date": 13,
  "day": 6,
  "full": "2018-01-13"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": true,
    "data": [],
    "date": {
      "year": 2018,
      "month": 1,
      "date": 13,
      "day": 6,
      "full": "2018-01-13"
    }
  }
}
          </span>
           
        </div>
      </div>
    </div>
    <div
      class="calendar-body-row"
    >
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 1,
  "date": 14,
  "day": 0,
  "full": "2018-01-14"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": true,
    "data": [],
    "date": {
      "year": 2018,
      "month": 1,
      "date": 14,
      "day": 0,
      "full": "2018-01-14"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 1,
  "date": 15,
  "day": 1,
  "full": "2018-01-15"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": true,
    "data": [],
    "date": {
      "year": 2018,
      "month": 1,
      "date": 15,
      "day": 1,
      "full": "2018-01-15"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 1,
  "date": 16,
  "day": 2,
  "full": "2018-01-16"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": true,
    "data": [],
    "date": {
      "year": 2018,
      "month": 1,
      "date": 16,
      "day": 2,
      "full": "2018-01-16"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 1,
  "date": 17,
  "day": 3,
  "full": "2018-01-17"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": true,
    "data": [],
    "date": {
      "year": 2018,
      "month": 1,
      "date": 17,
      "day": 3,
      "full": "2018-01-17"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 1,
  "date": 18,
  "day": 4,
  "full": "2018-01-18"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": true,
    "data": [],
    "date": {
      "year": 2018,
      "month": 1,
      "date": 18,
      "day": 4,
      "full": "2018-01-18"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 1,
  "date": 19,
  "day": 5,
  "full": "2018-01-19"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": true,
    "data": [],
    "date": {
      "year": 2018,
      "month": 1,
      "date": 19,
      "day": 5,
      "full": "2018-01-19"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 1,
  "date": 20,
  "day": 6,
  "full": "2018-01-20"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": true,
    "data": [],
    "date": {
      "year": 2018,
      "month": 1,
      "date": 20,
      "day": 6,
      "full": "2018-01-20"
    }
  }
}
          </span>
           
        </div>
      </div>
    </div>
    <div
      class="calendar-body-row"
    >
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 1,
  "date": 21,
  "day": 0,
  "full": "2018-01-21"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": true,
    "data": [],
    "date": {
      "year": 2018,
      "month": 1,
      "date": 21,
      "day": 0,
      "full": "2018-01-21"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 1,
  "date": 22,
  "day": 1,
  "full": "2018-01-22"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": true,
    "data": [],
    "date": {
      "year": 2018,
      "month": 1,
      "date": 22,
      "day": 1,
      "full": "2018-01-22"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 1,
  "date": 23,
  "day": 2,
  "full": "2018-01-23"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": true,
    "data": [],
    "date": {
      "year": 2018,
      "month": 1,
      "date": 23,
      "day": 2,
      "full": "2018-01-23"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 1,
  "date": 24,
  "day": 3,
  "full": "2018-01-24"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": true,
    "data": [],
    "date": {
      "year": 2018,
      "month": 1,
      "date": 24,
      "day": 3,
      "full": "2018-01-24"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 1,
  "date": 25,
  "day": 4,
  "full": "2018-01-25"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": true,
    "data": [],
    "date": {
      "year": 2018,
      "month": 1,
      "date": 25,
      "day": 4,
      "full": "2018-01-25"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 1,
  "date": 26,
  "day": 5,
  "full": "2018-01-26"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": true,
    "data": [],
    "date": {
      "year": 2018,
      "month": 1,
      "date": 26,
      "day": 5,
      "full": "2018-01-26"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 1,
  "date": 27,
  "day": 6,
  "full": "2018-01-27"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": true,
    "data": [],
    "date": {
      "year": 2018,
      "month": 1,
      "date": 27,
      "day": 6,
      "full": "2018-01-27"
    }
  }
}
          </span>
           
        </div>
      </div>
    </div>
    <div
      class="calendar-body-row"
    >
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 1,
  "date": 28,
  "day": 0,
  "full": "2018-01-28"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": true,
    "data": [],
    "date": {
      "year": 2018,
      "month": 1,
      "date": 28,
      "day": 0,
      "full": "2018-01-28"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 1,
  "date": 29,
  "day": 1,
  "full": "2018-01-29"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": true,
    "data": [],
    "date": {
      "year": 2018,
      "month": 1,
      "date": 29,
      "day": 1,
      "full": "2018-01-29"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 1,
  "date": 30,
  "day": 2,
  "full": "2018-01-30"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": true,
    "data": [],
    "date": {
      "year": 2018,
      "month": 1,
      "date": 30,
      "day": 2,
      "full": "2018-01-30"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 1,
  "date": 31,
  "day": 3,
  "full": "2018-01-31"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": false,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": true,
    "data": [],
    "date": {
      "year": 2018,
      "month": 1,
      "date": 31,
      "day": 3,
      "full": "2018-01-31"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 2,
  "date": 1,
  "day": 4,
  "full": "2018-02-01"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": true,
    "isNextFirstDay": true,
    "isToday": false,
    "isCurMonth": false,
    "data": [],
    "date": {
      "year": 2018,
      "month": 2,
      "date": 1,
      "day": 4,
      "full": "2018-02-01"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 2,
  "date": 2,
  "day": 5,
  "full": "2018-02-02"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": true,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": false,
    "data": [],
    "date": {
      "year": 2018,
      "month": 2,
      "date": 2,
      "day": 5,
      "full": "2018-02-02"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 2,
  "date": 3,
  "day": 6,
  "full": "2018-02-03"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": true,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": false,
    "data": [],
    "date": {
      "year": 2018,
      "month": 2,
      "date": 3,
      "day": 6,
      "full": "2018-02-03"
    }
  }
}
          </span>
           
        </div>
      </div>
    </div>
    <div
      class="calendar-body-row"
    >
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 2,
  "date": 4,
  "day": 0,
  "full": "2018-02-04"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": true,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": false,
    "data": [],
    "date": {
      "year": 2018,
      "month": 2,
      "date": 4,
      "day": 0,
      "full": "2018-02-04"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 2,
  "date": 5,
  "day": 1,
  "full": "2018-02-05"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": true,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": false,
    "data": [],
    "date": {
      "year": 2018,
      "month": 2,
      "date": 5,
      "day": 1,
      "full": "2018-02-05"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 2,
  "date": 6,
  "day": 2,
  "full": "2018-02-06"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": true,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": false,
    "data": [],
    "date": {
      "year": 2018,
      "month": 2,
      "date": 6,
      "day": 2,
      "full": "2018-02-06"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 2,
  "date": 7,
  "day": 3,
  "full": "2018-02-07"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": true,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": false,
    "data": [],
    "date": {
      "year": 2018,
      "month": 2,
      "date": 7,
      "day": 3,
      "full": "2018-02-07"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 2,
  "date": 8,
  "day": 4,
  "full": "2018-02-08"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": true,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": false,
    "data": [],
    "date": {
      "year": 2018,
      "month": 2,
      "date": 8,
      "day": 4,
      "full": "2018-02-08"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 2,
  "date": 9,
  "day": 5,
  "full": "2018-02-09"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": true,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": false,
    "data": [],
    "date": {
      "year": 2018,
      "month": 2,
      "date": 9,
      "day": 5,
      "full": "2018-02-09"
    }
  }
}
          </span>
           
        </div>
      </div>
      <div
        class="calendar-day-item"
      >
        <div>
          <span>
            {
  "year": 2018,
  "month": 2,
  "date": 10,
  "day": 6,
  "full": "2018-02-10"
} {
  "date": {
    "isPrevMonth": false,
    "isPrevLastDay": false,
    "isNextMonth": true,
    "isNextFirstDay": false,
    "isToday": false,
    "isCurMonth": false,
    "data": [],
    "date": {
      "year": 2018,
      "month": 2,
      "date": 10,
      "day": 6,
      "full": "2018-02-10"
    }
  }
}
          </span>
           
        </div>
      </div>
    </div>
    <div
      class="calendar-body-row"
    >
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
    </div>
  </div>
</div>
`;

exports[`Calendar component should match snapshot 1`] = `
<div
  class="calendar is-month"
>
  <div
    class="calendar-header"
  >
    <div
      class="calendar-header-left"
    />
    <div
      class="calendar-header-center"
    >
      <a
        class="calendar-control calendar-prev"
      >
        &lt;
      </a>
      <span
        class="calendar-header-date"
      >
        2018-01
      </span>
      <a
        class="calendar-control calendar-next"
      >
        &gt;
      </a>
    </div>
    <div
      class="calendar-header-right"
    />
  </div>
   
  <div
    class="calendar-week"
  >
    <div
      class="calendar-week__item"
    >
      
      Sun
    
    </div>
    <div
      class="calendar-week__item"
    >
      
      Mon
    
    </div>
    <div
      class="calendar-week__item"
    >
      
      Tue
    
    </div>
    <div
      class="calendar-week__item"
    >
      
      Wed
    
    </div>
    <div
      class="calendar-week__item"
    >
      
      Thu
    
    </div>
    <div
      class="calendar-week__item"
    >
      
      Fri
    
    </div>
    <div
      class="calendar-week__item"
    >
      
      Sat
    
    </div>
  </div>
   
  <div
    class="calendar-body"
  >
    <div
      class="calendar-body-row"
    >
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
    </div>
    <div
      class="calendar-body-row"
    >
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
    </div>
    <div
      class="calendar-body-row"
    >
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
    </div>
    <div
      class="calendar-body-row"
    >
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
    </div>
    <div
      class="calendar-body-row"
    >
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
    </div>
    <div
      class="calendar-body-row"
    >
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
      <div
        class="calendar-day-item"
      />
    </div>
    <div
      class="calendar-body-row"
    >
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
    </div>
  </div>
</div>
`;
